### PR TITLE
Cookbook Guide for Mirabella Genio Smart Bulbs

### PR DESCRIPTION
**Pull request by cryptelli**
## Description:
Cookbook guide for Mirabella Genio Smart Bulbs with configuration examples.

Co-Authored-By: MartinP <mplinuxgeek@users.noreply.github.com>

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>
**Pull request in [esphome-core](https://github.com/esphome/esphome-core) with C++ framework changes (if applicable):** esphome/esphome-core#<esphome-core PR number goes here>

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.

### CODE REVIEW

File: cookbook/mirabella-genio-bulb.rst [hunk 1]
```diff
@@ -0,0 +1,175 @@
+Mirabella Genio Bulbs
+=====================
+
+The Mirabella Genio is a Tuya based smart bulb sold by Kmart in Australia. 
+
+.. figure:: images/mirabella-genio-b22-rgbw.jpg
+    :align: center
+    :width: 50.0%
+
+Originally intended to be used with their companion app once flashed using `tuya-convert <https://github.com/ct-Open-Source/tuya-convert>`__ ESPHome generated firmware can be uploaded allowing you to control the bulbs via Home Assistant.
```
Comment (OttoWinter) [inline cookbook/mirabella-genio-bulb.rst]: We should have a quick note here about how to use tuya-convert with ESPHome. AFAIK the setup is a bit more difficult, as you need to first generate the bin and the flash it from tuya-convert. Doesn't have to be long and we can link to other tutorials.

File: cookbook/mirabella-genio-bulb.rst [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      name: mirabella_genio_cw_1
+      platform: ESP8266
+      board: esp01_1m
+
+      # Ensure the light turns on by default if the physical switch is actuated.
+      # https://esphome.io/components/esphome#esphome-on-boot
+      on_boot:
+        priority: 100 # Highest priority, ensures light turns on without delay.
+        then:
+          - light.turn_on: light
```
Comment (OttoWinter) [inline cookbook/mirabella-genio-bulb.rst:132]: So the physical switch on the device turns off the whole ESP?

File: cookbook/mirabella-genio-bulb.rst [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+    wifi:
+      ssid: 'WIFI'
+      password: 'WIFIPASS'
+
+    logger:
+
+    web_server:
+      port: 80
```
Comment (OttoWinter) [inline cookbook/mirabella-genio-bulb.rst]: Let's not enable the web server by default unless there's a specific reason to do so.

File: cookbook/mirabella-genio-bulb.rst [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    :align: center
+    :width: 50.0%
+
+Originally intended to be used with their companion app once flashed using `tuya-convert <https://github.com/ct-Open-Source/tuya-convert>`__ ESPHome generated firmware can be uploaded allowing you to control the bulbs via Home Assistant.
+
+Thanks to the `existing work <https://github.com/arendst/Sonoff-tasmota/wiki/Mirabella-Genio-Bulb>`__ by `@so3n <https://github.com/arendst/Sonoff-tasmota/wiki/Mirabella-Genio-Bulb>`__ which these adaptions created by `@cryptelli <https://community.home-assistant.io/u/cryptelli>`__ and `@bircoe <https://community.home-assistant.io/u/bircoe>`__ are based on.
+
+Dimmable Warm/Cool White Bulbs
+------------------------------
+
```
Comment (OttoWinter) [inline cookbook/mirabella-genio-bulb.rst:15]: Please add some comments here as to what this config is doing.

Can be quite short too, just mention that `esp8266_pwm` is used here to control the light, and how they're connected with IDs.

File: cookbook/mirabella-genio-bulb.rst [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+.. figure:: images/mirabella-genio-b22-rgbw.jpg
+    :align: center
+    :width: 50.0%
+
+Originally intended to be used with their companion app once flashed using `tuya-convert <https://github.com/ct-Open-Source/tuya-convert>`__ ESPHome generated firmware can be uploaded allowing you to control the bulbs via Home Assistant.
+
+Thanks to the `existing work <https://github.com/arendst/Sonoff-tasmota/wiki/Mirabella-Genio-Bulb>`__ by `@so3n <https://github.com/arendst/Sonoff-tasmota/wiki/Mirabella-Genio-Bulb>`__ which these adaptions created by `@cryptelli <https://community.home-assistant.io/u/cryptelli>`__ and `@bircoe <https://community.home-assistant.io/u/bircoe>`__ are based on.
+
+Dimmable Warm/Cool White Bulbs
```
Comment (OttoWinter) [inline cookbook/mirabella-genio-bulb.rst]: This section shows a monochromatic light? Let's call this section brightness-only or even monochromatic bulbs

File: cookbook/mirabella-genio-bulb.rst [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        name: "Mirabella Genio Smart Bulb"
+        id: light
+        output: output_component1
+
+    output:
+      - platform: esp8266_pwm
+        id: output_component1
+        pin: GPIO13
+
+Tunable Warm/Cool White Bulbs
```
Comment (OttoWinter) [inline cookbook/mirabella-genio-bulb.rst]: Let's not use `tunable` here, the word could mean anything here. Maybe something more like cold white + warm white bulbs

File: cookbook/mirabella-genio-bulb.rst [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+    light:
+      - platform: cwww
+        name: "Mirabella Genio Smart Bulb"
+        id: light
+        cold_white: output_daylight
+        warm_white: output_warm_white
+        cold_white_color_temperature: 6500 K
+        warm_white_color_temperature: 2700 K
+
+RGBW Colour Bulbs
```
Comment (OttoWinter) [inline cookbook/mirabella-genio-bulb.rst]:
```suggestion
RGBW Color Bulbs
```

American english :)

File: cookbook/mirabella-genio-bulb.rst [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+        green: output_green
+        blue: output_blue
+        white: output_white
+
+
+Addding to Home Assistant
+-------------------------
+
+You can now add your bulb to home assistant via the configurations page, look for 'ESPHome' under the Integrations option and click 'Configure'.
+
```
Comment (OttoWinter) [inline cookbook/mirabella-genio-bulb.rst:252]: If possible, add a screenshot here of what it looks like in HA - best with the RGB version and with the color picker open.

**Comment by cryptelli** [comment]
Boom 💥. Didn't intend on making more work for you! You've made very valid points which I'll address 👍.

File: cookbook/mirabella-genio-bulb.rst (hunk 2 above)
Comment (cryptelli) [reply cookbook/mirabella-genio-bulb.rst:132]: Sorry, not sure what you mean here 😕.

There is no physical switch apart from the one that controls the light, without the `on_boot` the light doesn't automatically turn on if the switch is toggled.

File: cookbook/mirabella-genio-bulb.rst (hunk 7 above)
Comment (cryptelli) [reply cookbook/mirabella-genio-bulb.rst]: :smile:

**Comment by OttoWinter** [comment]
Sorry for the long delay, I was quite busy the last few days.

Thanks for the detailed guide. The text looks good to me.

However, this guide should be created against the `current` branch (not next) as it's not a new esphome feature being documented, just a documentation update.

So to fix this, you need to go to the git repo

```bash
git checkout dev
git pull
git checkout -b mirabella-genio
# Copy changes here
git add .
git commit -m "Commit Message"
git push
```

(and see travis log, there are a few minor style issues in this PR that need to be addressed)

**Comment by cryptelli** [comment]
All good, likewise. I misunderstood the meaning of the next branch. I just changed the branch the pull request is based on above, hopefully that's suitable?

~~Regarding the travis log, as I'm fairly new to reStructuredText not sure how to address the line length issues in my lists?~~ Also the issue it has with links not being relative are in my code comments which won't be helpful to the end user if changed.

**Comment by cryptelli** [comment]
I've addressed most style errors, apart from the comments in my code blocks.

**Comment by OttoWinter** [comment]
Ok, thanks @cryptelli!

The branch you created the PR from was still `next` so commits that were not part of this PR were showing up (and all recent commits from next would go into current if I merged, which would break a bunch of things). So I reset the branch to current and cherry picked all commits that were part of this PR.

I also updated the contributing section to give some more info how to use branches in git (ideally you wouldn't create PRs from your current/next branch but from a new feature branch).

I will wait for the PR checks to pass and then merge it.

**Comment by cryptelli** [comment]
Sorry @OttoWinter... appreciate you merging this PR after all that!

The expanded documentation is much clearer. I'll be sure to follow the correct method on my next PR.

**Pull request merged**